src/TriangleNodesProvisioner.cpp

```diff
@@ -73,6 +77,179 @@ namespace blitzdg {
         buildMaps();
     }
 
+    GaussFaceContext2D TriangleNodesProvisioner::buildGaussFaceNodes(index_type NGauss) {
+        blitz::firstIndex ii;
+        blitz::secondIndex jj;
+        blitz::thirdIndex kk;
+
+        index_type Np = NumLocalPoints;
+
+        real_vector_type z(NGauss), w(NGauss);
+        real_vector_type face1r(NGauss), face2r(NGauss), face3r(NGauss);
+        real_vector_type face1s(NGauss), face2s(NGauss), face3s(NGauss);
+
+        // alpha = beta = 0, for Legendre polynomials.
+        Jacobi.computeJacobiQuadWeights(0, 0, NGauss - 1, z, w);
+
+        face1r = z; face2r = -z; face3r = 0.0*z - 1.0;
+        face1s = 0.0*z - 1.0; face2s = z; face3s = -z;
+
+        real_matrix_type V1(NGauss, Np), V2(NGauss, Np), V3(NGauss, Np);
+        
+        computeVandermondeMatrix(NOrder, face1r, face1s, V1);
+        computeVandermondeMatrix(NOrder, face2r, face2s, V2);
+        computeVandermondeMatrix(NOrder, face3r, face3s, V3);
+
+        real_matrix_type gaussFaceInterp1(NGauss, Np, ColumnMajorOrder()), 
+            gaussFaceInterp2(NGauss, Np, ColumnMajorOrder()), gaussFaceInterp3(NGauss, Np, ColumnMajorOrder());
+
+        real_matrix_type & invV = *Vinv, Drref = *Dr, Dsref = *Ds;
+        gaussFaceInterp1 = blitz::sum(V1(ii, kk)*invV(kk, jj), kk);
+        gaussFaceInterp2 = blitz::sum(V2(ii, kk)*invV(kk, jj), kk);
+        gaussFaceInterp3 = blitz::sum(V3(ii, kk)*invV(kk, jj), kk);
+
+    
+        real_matrix_type gaussInterpMatrix(NumFaces*NGauss, Np);
+        gaussInterpMatrix(Range(0,          NGauss - 1), Range::all()) = gaussFaceInterp1;
+        gaussInterpMatrix(Range(1*NGauss, 2*NGauss - 1), Range::all()) = gaussFaceInterp2;
+        gaussInterpMatrix(Range(2*NGauss, 3*NGauss - 1), Range::all()) = gaussFaceInterp3;
+
+        index_vector_type gmapM(NGauss*NumFaces*NumElements), gmapP(NGauss*NumFaces*NumElements);
+        gmapM = ii; gmapP = ii;
+
+        index_hashmap gbcMap;
```

Comment:
I would write this using an initializer list:
```
index_hashmap gbcMap = {
    {BCTag::Wall, std::vector<index_type>()},
    {BCTag::Dirichlet, std::vector<index_type>()},
    {BCTag::Neuman, std::vector<index_type>()},
    {BCTag::In, std::vector<index_type>()},
    {BCTag::Out, std::vector<index_type>()},
    {BCTag::Cyl, std::vector<index_type>()},
    {BCTag::Far, std::vector<index_type>()},
    {BCTag::Slip, std::vector<index_type>()}
};
```

Reply:
good point.